### DISCUSSION TITLE
Filter interface/virtual interface on assigned untagged vlan using API

### DISCUSSION THREAD

**Discussion by ziggekatten** [netbox-community/netbox#12319] [Help Wanted!]
It seems that it's not possible to filter on vlan_id or vlan_vid/group using the API. The swagger documentation shows no such option.

Am I right, or have I missed sometning here? I see references to that the feature was added in 2.4.7 (https://github.com/digitalocean/netbox/issues/2427), but those keys do not work in either of 3.2.9 or 3.4.4 that we have as running versions.

**Comment by mtinberg** [comment]
What data model are you having an issue with, the vlan model allows search by vid

/ipam/vlans/?vid=2


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: ziggekatten ***@***.***>
Sent: Friday, April 21, 2023 8:57 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Filter interface/virtual interface on assigned untagged vlan using API (Discussion #12319)


It seems that it's not possible to filter on vlan_id or vlan_vid/group using the API. The swagger documentation shows no such option.

Am I right, or have I missed sometning here? I see references to that the feature was added in 2.4.7 (#2427<https://github.com/netbox-community/netbox/issues/2427>), but those keys do not work in either of 3.2.9 or 3.4.4 that we have as running versions.

**Reply by ziggekatten** [reply]
As stated in the title: dcim.interfaces and virtualization.interfaces

**Comment by mtinberg** [comment]
Ah so, thanks.

So in my testing /dcim/interfaces/?vlan_id=12345 finds interfaces referencing VLAN record id 12345 /ipam/vlans/?id=12345 and /dcim/interfaces/?vlan=10 finds all interfaces with /ipam/vlans/?vid=10 which may be several different VLAN records in different Sites or VLAN Groups.  I didn't cross-check on /virtualization/interfaces but I'd expect it to work similarly (virt interfaces are different because they don't connect to physical cables).  If you are searching by VID it doesn't look like you can further filter by VLAN Group on the /dcim/interfaces endpoint but you can always do a two step lookup, get the record ID from /ipam/vlans/?vid=10&group=your_group&brief=1 then lookup /dcim/interfaces/?vlan_id=


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Comment by ziggekatten** [comment]
dcim/interfaces/?vlan_id=xxxx works on physical interface. On virtualization/interfaces/?vlan_id=xxxx always return all interfaces, thus indicating that there is an difference between physical and virtual interfaces in terms of how filters are implemented. Bummer if that is the case.

**Reply by kkthxbye-code** [reply]
Both filters work just fine for me:

https://demo.netbox.dev/api/virtualization/interfaces/?vlan_vid=1337

https://demo.netbox.dev/api/virtualization/interfaces/?vlan_id=64

**Reply by kkthxbye-code** [reply]
Here, it was added in v3.4.8:

https://github.com/netbox-community/netbox/releases/tag/v3.4.8

> https://github.com/netbox-community/netbox/issues/12007 - Enable filtering of VM Interfaces by assigned VLAN

**Reply by abhi1693** [reply]
This feature was added in v3.4.8, so if you are using an old version you'd get all interfaces as the filter doesn't exist in them.

**Comment by ziggekatten** [comment]
Time for an upgrade of Netbox then, since I'm at older versions.

**Comment by mtinberg** [comment]
That's the easiest kind of bugfix, the work is already done! 😉


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
